### PR TITLE
handle return val of fwrite() to pacify compiler warn

### PR DESCRIPTION
**Pull request by andrewhsu**
Getting compiler warnings on gcc 4.4.7:
```
+ make tini-static
Scanning dependencies of target tini-static
[100%] Building C object CMakeFiles/tini-static.dir/src/tini.c.o
cc1: warnings being treated as errors
/go/tini/src/tini.c: In function 'print_license':
/go/tini/src/tini.c:211: error: ignoring return value of 'fwrite', declared with attribute warn_unused_result
make[3]: *** [CMakeFiles/tini-static.dir/src/tini.c.o] Error 1
make[2]: *** [CMakeFiles/tini-static.dir/all] Error 2
make[1]: *** [CMakeFiles/tini-static.dir/rule] Error 2
make: *** [tini-static] Error 2
```

See https://github.com/docker/docker/issues/28273 for background.

For reference:
```
# gcc --version
gcc (GCC) 4.4.7 20120313 (Red Hat 4.4.7-17)
```

### CODE REVIEW

**Comment by andrewhsu** [comment]
@crosbymichael fyi i've checked for return value instead of just ignoring the return val in accordance with [man page](https://linux.die.net/man/3/fwrite)

**Comment by krallin** [comment]
Hey @andrewhsu, thanks a lot for bringing this up! 

I'd like to avoid having to many `exit()` points in Tini, so do you think we could just skip that part and do nothing in the error branch? What about:

```
void print_license(FILE* const file) {
	if(LICENSE_len > fwrite(LICENSE, sizeof(char), LICENSE_len, file)) {
		// Don't handle this error for now, since parse_args won't care
		// about the return value. We do need to check it to compile with
		// older glibc, though.
		// See: https://gcc.gnu.org/bugzilla/show_bug.cgi?id=25509
		// See: http://sourceware.org/bugzilla/show_bug.cgi?id=11959
	}
}
```

(I also think it's worth adding a comment around just so we don't accidentally break this for you later!)

This appears to compile fine for me, but please let me know if it does with your older gcc / libc (I do believe the issue you're running into is due to glibc, not gcc: http://sourceware.org/bugzilla/show_bug.cgi?id=11959)

To summarize:

- Thanks again for this PR!
- Let me know if the proposed change fix looks good for you, and if it does, please amend the PR and I'll merge.
- Please also feel free to add yourself to the contributors list (update the README template under `tpl` then rebuild to update the actual README — I'm happy to do this for you if needed).

---

To be clear: in the longer term I agree Tini should exit with 1 here if `fwrite` fails, but this should be handled by `parse_args`, not `print_license`, as it's a problem in other places as well (`print_usage` also ignores return values for `fprintf`). This should not hold up this fix you though, hence why I suggest we just ignore the error for now!

Thanks!

**Comment by crosbymichael** [comment]
Update with the comment

**Comment by krallin** [comment]
Thanks! I'll merge this now.
